### ISSUE TITLE
Do not depend on sun packages

### ISSUE THREAD

**Issue by theangrydev** [bodar/totallylazy#22]
There are some references to sun packages (e.g. sun.reflect) in totallylazy.

These references should be removed, since you should not be depending on them, as explained here:
http://www.oracle.com/technetwork/java/faq-sun-packages-142232.html

**Comment by danielbodart** [comment]
Was there a feature you needed on a non Oracle JVM? Maybe Android?

**Comment by theangrydev** [comment]
Nope, I don't have a particular use case. Someone at work pointed this out today and they were quite startled by it, so I thought I'd raise an issue for it

**Comment by danielbodart** [comment]
Okay well I'm planning to remove them for the Java 9 version (no choice with the new modules system) but then a fair amount of sun stuff become official then so it's easy in newer versions. 

I won't be back porting to Java 7 as there is just way to much stuff missing.
